### ISSUE TITLE
reltuples for external partition are now estimated as 0

### ISSUE THREAD

**Issue by gcaragea** [greenplum-db/gpdb#395]
This is causing a failure in our regression tests. The reltuples for an external partition was previously estimated at 1M, now it is estimated at 0. This was introduced by the commit that brought ANALYZE from upstream: https://github.com/greenplum-db/gpdb/commit/f995c59f349917a99407e3811d9b317d5d77b257

Way to reproduce the problem:
```
drop table if exists pt_ext;
CREATE TABLE pt_ext
(
  col1 int,
  col2 decimal,
  col3 text,
  col4 bool
)
distributed by (col1)
partition by list(col2)
(
	partition part1 values(1,2,3,4,5,6,7,8,9,10),
	partition part2 values(11,12,13,14,15,16,17,18,19,20),
	partition part3 values(21,22,23,24,25,26,27,28,29,30),
	partition part4 values(31,32,33,34,35,36,37,38,39,40),
	partition part5 values(41,42,43,44,45,46,47,48,49,50)
);

insert into pt_ext select i,i,'test',true from generate_series(1,50) i;
create temp table tmp as select * from pt_ext where col1 < 11;
\! rm /tmp/exttab_list
copy tmp to '/tmp/exttab_list' csv;

select hostname from gp_segment_configuration where content=-1;
-- REPLACE HOST WITH THE STRING RETURNED BY THE PREVIOUS QUERY
create readable external table ret(like pt_ext) location('file://HOST/tmp/exttab_list') format 'csv';
alter table pt_ext exchange partition part1 with table ret without validation;
drop table ret;

analyze pt_ext_1_prt_part1;
analyze verbose pt_ext_1_prt_part1;
analyze pt_ext;
select reltuples from pg_class where relname='pt_ext_1_prt_part1';
```

reltuples is now returned as 0. Before the fix, it was returning 1M.

**Comment by atris** [comment]
PR #385 should fix that for some cases

**Comment by hlinnaka** [comment]
PR #385 might indeed help with the planner-side of things. We still shouldn't run ANALYZE on an external table, though. Created PR #617 to stop that from happening.

**Comment by hlinnaka** [comment]
Fixed by PR #617. (Sorry for the long response time, I only noticed this issue in the tracker yesterday)
